src/lib/frontend/frontend.ml

```diff
@@ -143,10 +143,10 @@ module Make(SAT : Sat_solver_sig.S) : S with type sat_env = SAT.t = struct
   let process_decl print_status used_context ((env, consistent, dep) as acc) d =
     try
       match d.st_decl with
-      | Push -> SAT.push env, consistent, dep
-      | Pop ->
+      | Push n -> SAT.push env n, consistent, dep
+      | Pop n ->
         (* Reset the constitency for each pop command *)
-        SAT.pop env, true, dep
+        SAT.pop env n, true, dep
```

Comment:
I haven't noticed it during the first review: is it correct to always return `dep` ?

Reply:
No i don't think it's correct indeed. Since we force the reset of constintency for `pop` command, should we retrieve the `dep` that was computed before the n assertion levels we pushed ?

Reply:
Yes, I think we should do this

Reply:
Done. I also fixed the consistency issue retrieving the consistency of the previous assertion level